core/src/impl/Kokkos_Command_Line_Parsing.hpp

```diff
@@ -62,6 +63,8 @@ void warn_deprecated_environment_variable(std::string deprecated,
 void warn_deprecated_command_line_argument(std::string deprecated);
 void warn_deprecated_command_line_argument(std::string deprecated,
                                            std::string use_instead);
+void warn_not_recognized_command_line_argument(std::string not_recognized);
+void do_not_warn_not_recognized_command_line_argument(std::regex ignore);
```

Comment:
An alternative here would be to taken the prefix the option not to warn about start with (take `std::string` instead of `std::regex`)
Let me know which you prefer

Reply:
I do like the regex. I think it is more flexible than passing a string.
Do you plan on adding a test?

Reply:
Added

Reply:
So extending the unit test 
```diff
@@ -290,6 +297,21 @@ TEST(defaultdevicetype, cmd_line_args_unrecognized_flag) {
   Kokkos::Impl::parse_command_line_arguments(cla.argc(), cla.argv(), settings);
   captured = ::testing::internal::GetCapturedStderr();
   EXPECT_TRUE(captured.empty() && !settings.has_num_threads()) << captured;
+  EXPECT_EQ(cla.argc(), 1);
+  EXPECT_STREQ(cla.argv()[0], "--kokko-num-threads=4");
+
+  Kokkos::Impl::do_not_warn_not_recognized_command_line_argument(
+      std::regex{"^--kokkos-extension.*"});
+  cla = {{
+      "--kokkos-extension-option=value",  // user explicitly asked not to warn
+                                          // about that prefix
+  }};
+  ::testing::internal::CaptureStderr();
+  Kokkos::Impl::parse_command_line_arguments(cla.argc(), cla.argv(), settings);
+  captured = ::testing::internal::GetCapturedStderr();
+  EXPECT_TRUE(captured.empty()) << captured;
+  EXPECT_EQ(cla.argc(), 1);
+  EXPECT_STREQ(cla.argv()[0], "--kokkos-extension-option=value");
 }

 TEST(defaultdevicetype, env_vars_num_threads) {
```
exposed an issue with the current implementation
```
[ RUN      ] defaultdevicetype.cmd_line_args_unrecognized_flag
/Users/qdi/CLionProjects/kokkos/core/unit_test/TestParseCmdLineArgsAndEnvVars.cpp:313: Failure
Expected equality of these values:
  cla.argc()
    Which is: 0
  1
[  FAILED  ] defaultdevicetype.cmd_line_args_unrecognized_flag (1 ms)
```
We are removing all flags that are prefixed with `--kokkos` regardless of whether we handled them or not...

Reply:
we should mark this as a feature ...

Reply:
Fixed